frontend/src/Onboarding/OnboardingInfoScreen.tsx

```diff
@@ -354,7 +367,12 @@ class OnboardingScreenComponent extends React.Component<
 
           {textFieldStr.length !== 0 && !!year && !!gradYear ? (
             <Link
-              to={!!this.state.major ? "/completedCourses" : "/signup"}
+              to={{
+                pathname: !!this.state.major
+                  ? "/completedCourses"
+                  : this.onBoardLink(fromOnBoardingGuest),
```

Comment:
You can actually chain terenary operators like this
!!this.state.major  ? "/completedCourses" :  fromOnBoardingGuest ? "/home" : "/signup"